components/s3_bucket/main.tf

```diff
@@ -62,6 +68,26 @@ resource "aws_s3_bucket" "bucket" {
     enabled = var.versioning || var.replication_target != null
   }
 
+  dynamic "lifecycle_rule" {
+    for_each = var.temporary_paths
+
+    content {
+      id      = md5(lifecycle_rule.value)
+      prefix  = lifecycle_rule.value
+      enabled = true
+
+      # Delete files in this path after two weeks:
+      expiration {
+        days = 14
+      }
+
+      # If versioned, delete non-current versions at this path after 3 months:
```

Comment:
What does this mean?

Reply:
We have versioning enabled on our production bucket, so when we delete a file (or overwrite it, or whatnot) S3 keeps historical copies in case we need to roll back. This gives us extra peace of mind that we never lose user data from a bug! However, it'd also mean we'd keep these temporary files indefinitely unless we add a rule to remove those old versions too.